src/components/Appointment/BookAppointmentForm/SecondStep/index.tsx

```diff
@@ -47,109 +50,153 @@ const SecondStepAppointment = ({
+                            type="radio"
+                            value={doc.doctor.id}
+                            checked={selectedDoctor == doc.doctor.id}
+                            onChange={(e) =>
+                              handleDoctorChange(e.target.value, onChange)
+                            }
+                            errors={errors}
+                          />
+                        )}
                       />
-                    )}
-                  />
-                  <DoctorImg
-                    src={defaultDoctorPhoto}
-                    alt={doctor}
-                    width={32}
-                    height={32}
-                  />
-                  <DoctorName>{doc.name}</DoctorName>
-                </DoctorItemInfo>
-                <DoctorItemInfo>{doc.speciality}</DoctorItemInfo>
-                <DoctorItemInfo> {doc.located}</DoctorItemInfo>
-              </DoctorItem>
-            </label>
-          ))
-        ) : (
-          <p> {t('BookAppointment.noDoctors')}</p>
-        )}
-      </DoctorsList>
-      {visibledoctorsLists.length > page + 1 && (
-        <LoadMoreBtn onClick={handleClickLoadMore}>
-          {' '}
-          {t('BookAppointment.loadMore')}
-        </LoadMoreBtn>
+                      {doc.doctor.photoUrl ? (
+                        <DoctorImg
+                          src={doc.doctor.photoUrl}
+                          alt={doctor}
+                          width={32}
+                          height={32}
+                        />
+                      ) : (
+                        <DoctorImg
+                          src={defaultDoctorPhoto}
+                          alt={doctor}
+                          width={32}
+                          height={32}
+                        />
+                      )}
+
+                      <DoctorName>
+                        {doc.doctor.firstName}, {doc.doctor.lastName}
+                      </DoctorName>
+                    </DoctorItemInfo>
+
+                    <DoctorItemInfo>
+                      {memoizedGetSpecializationLabel(
+                        doc.doctor.specialization
+                      )}
+                    </DoctorItemInfo>
+                    <DoctorItemInfo>
+                      {' '}
+                      {doc.doctor.city}, {doc.doctor.country}
+                    </DoctorItemInfo>
+                  </DoctorItem>
+                </label>
+              ))
+            ) : (
+              <p> {t('BookAppointment.noDoctors')}</p>
+            )}
+          </DoctorsList>
+          {filtered.length >= limit && (
+            <LoadMoreBtn onClick={handleLoadMore}>
+              {' '}
+              {t('BookAppointment.loadMore')}
+            </LoadMoreBtn>
+          )}
+          <FormFooter>
+            <BntWrapper>
+              <StepBtn
+                onClick={() => {
+                  setSelectedDoctor(null);
+                  setStep(false);
```

Comment:
if more then one function move to separate function

Reply:
fix. moved to a separate function